### DISCUSSION TITLE
pypowerwall background color

### DISCUSSION THREAD

**Discussion by Craig-G-NZ** [jasonacox/Powerwall-Dashboard#363] [General]
Not sure if ive got this right or not. I had to stuff around with doing a manual install on a new server i setup. one thing i cannot work out is why the background color on the animated graphics is white instead of black and the graphic seems to scroll where im sure it never use to when i did an automated setup?

heres what it looks like directly in tesla web front end
![image](https://github.com/jasonacox/Powerwall-Dashboard/assets/92700720/4ff63aba-e2aa-4f70-a808-3c16cd472409)

heres what i see in my powerwall dashboard
![image](https://github.com/jasonacox/Powerwall-Dashboard/assets/92700720/6566c6d0-4818-409f-b892-37d463853b11)

anyone got any ideas on how to make the dashboard look like the tesla front end or clean it up with a black background and remove the scrolling etc?

**Comment by jasonacox** [comment]
The pypowerwall is updating CSS to change the color (set by PW_STYLE) which by default should be "clear" which is what you want for the Grafana iframe panel.

A few questions to provide context for help:
* Why did you need to use the manual setup?
* Did you set up all the needed environmentals for Grafana (e.g. `GF_PANELS_DISABLE_SANITIZE_HTML=true` - see https://github.com/jasonacox/Powerwall-Dashboard/blob/main/grafana.env.sample)?
* Are you using the standar [dashboard.json](https://github.com/jasonacox/Powerwall-Dashboard/blob/main/dashboards/dashboard.json)?

**Reply by Craig-G-NZ** [reply]
Hi Jason,

Originally i was using truenas, and an ubuntu vm that was hosting docker. i moved away from that to truenas scale but then found issues with my reverse proxy and after spending a lot of time moved completely away from truenas.

i am now running centos, and using podman rather than docker. so i had to do the manual configuration with that.

Yes i have GF_PANELS_DISABLE_SANITIZE_HTML=true in my grafana.env file

Yes to using the standard dashboard.json

**Reply by jasonacox** [reply]
Thanks @Craig-G-NZ , 

Some ideas:

1)  I wonder if the pypowerwall container somehow lost access to the local `/web/` assets which includes the `clear.js` file needed to reset the background.  Are you able to see http://localhost:8675/example.html (replace localhost with the IP/name you use to see Grafana)?

    It should render something like this:
 
<img width="470" alt="image" src="https://github.com/jasonacox/Powerwall-Dashboard/assets/836718/9ccd5b14-c10d-453c-8ed8-22d0ac28facd">

2) Check your browser JavaScript console to see if there are any errors (on Chrome that is View > Developer > JavaScript Console).

3) Can you verify the versions of the containers you are running?  It would be interesting to see if `verify.sh` works.

**Reply by Craig-G-NZ** [reply]
this is what i see when i go to the example.html

![image](https://github.com/jasonacox/Powerwall-Dashboard/assets/92700720/e9eb81a1-b73a-4ec9-a712-1ce63a21ae0d)

**Reply by Craig-G-NZ** [reply]
these eventually came up in the console

Firefox can’t establish a connection to the server at wss://powerwall.mydomain.com/api/live/ws. [transport_websocket.js:32:30](webpack://grafana/yarn/cache/centrifuge-npm-4.0.1-48223b9b1e-5cfa25d268.zip/node_modules/centrifuge/build/transport_websocket.js)
The connection to wss://powerwall.mydomain.com/api/live/ws was interrupted while the page was loading.

**Reply by Craig-G-NZ** [reply]
this is what i see for the powerwall graphics

![image](https://github.com/jasonacox/Powerwall-Dashboard/assets/92700720/62ca123c-d43c-46c2-8400-37a1ce28f8ae)

**Reply by Craig-G-NZ** [reply]
if i add PW_STYLE=black i get this.

![image](https://github.com/jasonacox/Powerwall-Dashboard/assets/92700720/4d31ea90-97c1-4ef2-9690-084c1ca26d4f)

**Reply by jasonacox** [reply]
What happens when you try `PW_STYLE=clear`?  Also, try to clear the browser cache (or use another browser like chrome to confirm). The one example above is missing the firmware version which is odd. It shows up in some of your examples.

**Reply by Craig-G-NZ** [reply]
same white background with clear in firefox and edge.

![image](https://github.com/jasonacox/Powerwall-Dashboard/assets/92700720/ebc6a9a1-559c-42d6-be46-67ef4f1e1791)

**Reply by jasonacox** [reply]
Thanks for checing.  It also looks like the Javascript resize function isn't working either (I think those are scroll bars?). Hum...  I'm running out of ideas. I can't replicate it unfortunately. Could we check to see if Grafana is set up correctly?

* Make sure you are using version 9.1.2
* Make sure all the [environmentals](https://github.com/jasonacox/Powerwall-Dashboard/blob/main/grafana.env.sample) are loading, including the [plug-ins](https://github.com/jasonacox/Powerwall-Dashboard/blob/164f7595993a51ca38f1b1dd4af1352581e2f5d7/grafana.env.sample#L3).
* Can you run `docker logs grafana` and `docker logs pypowerwall` to see if there are any errors showing there?

**Reply by jasonacox** [reply]
Hum... did you happen to use the latest grafana (version 9.4.14)?  In my testing, I'm seeing some of the issue you see.  See https://github.com/jasonacox/Powerwall-Dashboard/discussions/371.

**Reply by jasonacox** [reply]
You might try this to see if it helps with your case: https://github.com/jasonacox/Powerwall-Dashboard/discussions/371#discussioncomment-7282312

**Reply by Craig-G-NZ** [reply]
Hi Jason. Sorry I didn't get back you. Looks like I'm on 10.1.4 version of grafana. I'll take a look at that other issue and see if that fixes my issue

**Reply by jasonacox** [reply]
That's helpful!  We pinned the project to version 9.1.2 because Grafana tends to create a lot of breaking changes.  It seems like there are some reported bugs in 10.x.x that may not make sense for us to accommodate just yet, but I think we may be able to move to 9.4.14.

**Reply by Craig-G-NZ** [reply]
Hey Jason, i just tested with "PW_STYLE=grafana-dark" and updated the grafana contianer. looks great :)

![image](https://github.com/jasonacox/Powerwall-Dashboard/assets/92700720/930d58e1-84e6-4c3e-b08b-c5d91921a4cc)

i noted there is still a scroll bar but its not that noticeable so im good with it now.

**Reply by jasonacox** [reply]
Thanks @Craig-G-NZ !  We are pushing a change to update the default install to include that setting.

As to the scroll bars, I can't replicate that.  Are you on Grafana 10.1.4 or 9.4.14? Also, what browser are you using?

Thanks for your help!

**Reply by Craig-G-NZ** [reply]
Hi there, I'm on 10.1.5 now and this is how it looks when scrolling as low as i can go. like i mentioned above its not a big deal and I'm not worried about it.

![image](https://github.com/jasonacox/Powerwall-Dashboard/assets/92700720/44e5c6bf-3e84-4a52-88bf-20fac985bf52)

I try to keep my containers up to date so my version number will continue to get further away from what you are using. I wouldn't stress too much about that scroll bar.

**Reply by jasonacox** [reply]
Understood. Thanks!
